source/magnifier/utils/types.py

```diff
@@ -0,0 +1,135 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2025 NV Access Limited, Antoine Haffreingue
+# This file may be used under the terms of the GNU General Public License, version 2 or later, as modified by the NVDA license.
+# For full terms and any additional permissions, see the NVDA license file: https://github.com/nvaccess/nvda/blob/master/copying.txt
+"""
+Types used in the magnifier module.
+"""
+
+from typing import NamedTuple
+from enum import Enum
+from utils.displayString import DisplayStringStrEnum
+from gettext import pgettext
+
+
+class MagnifierParams(NamedTuple):
+	"""Named tuple representing magnifier parameters for initialization"""
+
+	zoomLevel: float
+	filter: str
+	fullscreenMode: str
+
+
+class Direction(Enum):
+	"""Direction for zoom operations"""
+
+	IN = True
+	OUT = False
+
+
+class MagnifierAction(DisplayStringStrEnum):
+	"""Actions that can be performed with the magnifier"""
+
+	ZOOM_IN = "zoom_in"
+	ZOOM_OUT = "zoom_out"
+	TOGGLE_FILTER = "toggle_filter"
+	CHANGE_FULLSCREEN_MODE = "change_fullscreen_mode"
+	START_SPOTLIGHT = "start_spotlight"
```

Comment:
I don't think this needs to be a StrEnum

```suggestion
class MagnifierAction(DisplayStringEnum):
	"""Actions that can be performed with the magnifier"""

	ZOOM_IN = auto()
	ZOOM_OUT = auto()
	TOGGLE_FILTER = auto()
	CHANGE_FULLSCREEN_MODE = auto()
	START_SPOTLIGHT = auto()
```